src/sentry/grouping/ingest.py

```diff
@@ -456,32 +477,37 @@ def record_hash_calculation_metrics(
             "primary_config": primary_config["id"],
             "secondary_config": secondary_config["id"],
         }
-        current_values = primary_hashes.hashes
-        secondary_values = secondary_hashes.hashes
-        hashes_match = current_values == secondary_values
 
-        if hashes_match:
-            tags["result"] = "no change"
-        else:
-            shared_hashes = set(current_values) & set(secondary_values)
-            if len(shared_hashes) > 0:
-                tags["result"] = "partial change"
+        # If the configs are the same, *of course* the values are going to match, so no point in
+        # recording a metric
+        #
+        # TODO: If we fix the issue outlined in https://github.com/getsentry/sentry/pull/65116, we
+        # can ditch both this check and the logging below
+        if tags["primary_config"] != tags["secondary_config"]:
+            current_values = primary_hashes.hashes
+            secondary_values = secondary_hashes.hashes
+            hashes_match = current_values == secondary_values
+
+            if hashes_match:
+                tags["result"] = "no change"
             else:
-                tags["result"] = "full change"
-
-        metrics.incr("grouping.hash_comparison", tags=tags)
-
-        # TODO: This is temporary, just until we can figure out how we're recording a hash
-        # comparison metric showing projects calculating both primary and secondary hashes using the
-        # same config
-        if primary_config["id"] == secondary_config["id"]:
-            logger.info(
-                "Equal primary and secondary configs",
-                extra={
-                    "project": project.id,
-                    "primary_config": primary_config["id"],
-                },
-            )
+                shared_hashes = set(current_values) & set(secondary_values)
+                if len(shared_hashes) > 0:
+                    tags["result"] = "partial change"
+                else:
+                    tags["result"] = "full change"
+
+            metrics.incr("grouping.hash_comparison", tags=tags)
+
+        else:
+            if not _config_update_happened_recently(project, 30):
```

Comment:
would it be possible to add a test to cover this branch?

Reply:
Sure. Didn't think it was worth the time for a log that's likely to get ripped out once we can confirm that relay caching is the reason behind the weird data, but I can if you think it's important. Given the lifespan of the check, do you still?

Reply:
(Also, if we can confirm that, we'll switch from using the relay-provided primary config to using the one on the project, at which point the entire `else` will become a logical impossibility.)

Reply:
just being _extra_ careful  :D